user_sync/rules.py

```diff
@@ -347,9 +379,10 @@ def try_and_remove_from_org(user_key):
                 if (not owning_organization_info.is_dashboard_users_loaded() or owning_organization_info.get_dashboard_user(user_key) != None):
                     self.logger.info('Removing user for user key: %s', user_key)
                     id_type, username, domain = self.parse_user_key(user_key)
+                    do_delete_user = self.delete_after_remove_from_org and id_type != user_sync.identity_type.ADOBEID_IDENTITY_TYPE
```

Comment:
The `and` is not needed here, `remove_from_org` will do the right thing on Adobe IDs without a complaint, even if you say `do_delete_user=True`.